cmake/CompilerWarnings.cmake

```diff
@@ -109,6 +109,16 @@ function(mamba_target_add_compile_warnings target)
         -Wuninitialized
     )
 
+    # It seems that there are some wrong flags added to CXXFLAGS during `conda-build`. And then
+    # these flags are used to compile our source code. It might be a bug in
+    # https://github.com/conda-forge/clang-compiler-activation-feedstock.
```

Comment:
I rewrote the comment and added more details.

**Note, this is not caused by my PR, it already happens in `main`**
But because we use warnings, not errors, no one noticed:
https://github.com/mamba-org/mamba/actions/runs/11935345040/job/33266465568#step:7:1202

Unfortunately, I couldn't understand exactly why this happens.
But I noticed `INFO: activate_clangxx_osx-64.sh made the following environmental changes:` a few lines above.
And then we have lines showing erroneous changes in `CXXFLAGS`.
That's why I think it happens in the mentioned lib.

If someone wants to dig deeper, that would be great.